internal/server/iigointernal/executive.go

```diff
@@ -0,0 +1,148 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+)
+
+type executive struct {
+	ID               shared.ClientID
+	clientPresident  roles.President
+	budget           shared.Resources
+	speakerSalary    shared.Resources
+	RulesProposals   []string
+	ResourceRequests map[shared.ClientID]shared.Resources
+}
+
+// returnSpeakerSalary returns the salary to the common pool.
+func (e *executive) returnSpeakerSalary() shared.Resources {
+	x := e.speakerSalary
+	e.speakerSalary = 0
+	return x
+}
+
+// Get rule proposals to be voted on from remaining islands
+// Called by orchestration
+func (e *executive) setRuleProposals(rulesProposals []string) {
+	e.RulesProposals = rulesProposals
+}
+
+// Set approved resources request for all the remaining islands
+// Called by orchestration
+func (e *executive) setAllocationRequest(resourceRequests map[shared.ClientID]shared.Resources) {
+	e.ResourceRequests = resourceRequests
+}
+
+// Get rules to be voted on to Speaker
+// Called by orchestration at the end of the turn
+func (e *executive) getRuleForSpeaker() string {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.PickRuleToVote(e.RulesProposals)
+	return result
+}
+
+// Send Tax map all the remaining islands
+// Called by orchestration at the end of the turn
+func (e *executive) getTaxMap(islandsResources map[shared.ClientID]shared.Resources) map[shared.ClientID]shared.Resources {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.SetTaxationAmount(islandsResources)
+	return result
+}
+
+// broadcastTaxation broadcasts the tax amount decided by the president to all island still in the game.
+func (e *executive) broadcastTaxation(islandsResources map[shared.ClientID]shared.Resources) {
+	e.budget -= serviceCharge
+	taxAmountMap := e.getTaxMap(islandsResources)
+	for _, v := range getIslandAlive() {
+		d := baseclient.Communication{T: baseclient.CommunicationInt, IntegerData: int(taxAmountMap[shared.ClientID(int(v))])}
+		data := make(map[int]baseclient.Communication)
+		data[TaxAmount] = d
+		communicateWithIslands(shared.TeamIDs[int(v)], shared.TeamIDs[e.ID], data)
+	}
```

Comment:
Why was communication designed like this? Would it not be easier to have specific methods for each type of communcation. I.e. a recieveTaxAmount() function in the base client.

With the enum-communication map and the receiveCommuncation() function, you're likely to end up with a large function with a lot of branches inside of it depending on the type of communication.

Reply:
I understand your concerns and I would agree with you if there were a smaller range of possible communications. From a design perspective, I would argue in favour of the recieveCommunication() function as this will be written by individual islands, which gives them the flexibility to save and consider only the information that they want to save and consider. 

Currently, there are 11 types of communication from IIGO which may increase post-MVP. None of these communications are **necessary for the game to function**. At a bare minimum, to avoid breaking the rules, a **compliant** island only needs to save a local copy of TaxAmount and AllocationAmount so that it contributes/deducts what it is allowed to from the common pool. The purpose of the majority of these communications is to help islands with their opinion formation to make choices on rule votes and elections.

For whatever reason, teams may decide that they do not want to consider whether the Speaker correctly performed the vote (SpeakerBallotCheck) in their opinion formation of the Speaker island. If they do not have a corresponding function to receiveSpeakerBallotCheck, we would have to do additional checks on the server side to avoid problems. Thus, every team would be forced to have a corresponding function for each type of communication regardless of whether they will use that information or not. This would result in a lot of 'junk' functions in their island code.

Reply:
Okay, that does seem well reasoned. For clarity however, could you name the enum inside the utilities.go file and use that type in the recieveCommunication function. Currently the map just goes from int to Communication which can be a bit confusing

Reply:
Sure thing! Done by @Neelesh99 !